upt/util/parser_common.py

```diff
@@ -0,0 +1,117 @@
+                                                 usage=self.__class__.usage)
+
+        if login_options:
+            self.argparser.add_argument("--init",
+                                        nargs=0,
+                                        action=self.__init_action(),
+                                        help="Initialize data",)
+            self.argparser.add_argument("-l",
+                                        "--login",
+                                        action="store_true",
+                                        help="Login before parse",)
+        self.argparser.add_argument("-i",
+                                    "--inplace",
+                                    action="store_true",
+                                    help="Create Tests inplace",)
+        self.argparser.add_argument("-u",
+                                    "--url",
+                                    nargs=1,
+                                    help="Custom task url",)
+        self.argparser.add_argument("task",
+                                    nargs="*",
+                                    help=argparse.SUPPRESS)
+        self.driver = None
+        self.driver_options = driver_options
+        self.login_options = login_options
+
+    def __init_action(self):
+        class MyAction(argparse.Action):
+            def __init__(obj, option_strings, dest, **kwargs):
+                super(MyAction, obj).__init__(option_strings, dest, **kwargs)
+
+            def __call__(obj, parser, namespace, values, option_string=None):
+                self.initialize()
+                sys.exit()
+        return MyAction
+
+    def initialize(self):
+        login = LoginManager(self.__class__.name)
+        login.get_auth()
+        login.write()
+
+    def url_finder(self, task):
+        raise Exception("No url_finder function")
+
+    def placer(self, task):
+        return f"/{self.__class__.name}/{task}"
+
+    def sampler(self, elements):
+        return sample_common.Sampler.even_odd(elements)
+
+    def login(self, url, username, password, submit, checker):
+        logger.info("Trying to login")
+        login = LoginManager(self.__class__.name)
+        user, pwd = login.read_auth()
+
+        self.driver.get(url)
+        if not self.driver_options.get("nostrategy") is False:
+            wait_until(self.driver, *username)
+
+        user_box = self.driver.find_element(*username)
+        user_box.send_keys(user)
+        pass_box = self.driver.find_element(*password)
+        pass_box.send_keys(pwd)
+
+        submit_btn = self.driver.find_element(*submit)
+        submit_btn.click()
+
+        checker(self.driver)
+
+    def parse(self, args):
+        args = self.argparser.parse_args(args)
+
+        url = args.url[0] if args.url else self.url_finder(args.task)
+        path = "./" if args.inplace or args.url else InitParser().get_path(self.placer(args.task), makedir=True)
```

Comment:
Pass `args.task` as `*args.task`. Consider other parsers too.
Also, handle the exception created by parsers.